dataapi/bool.go

```diff
@@ -0,0 +1,43 @@
+package dataapi
+
+// Bool is a DataItem that wraps a bool primitive
+type Bool struct {
+	*BaseDataItem
+	value  bool
+	locked bool
+}
+
+func NewBool(b bool) *Bool {
+	return &Bool{
+		BaseDataItem: NewBaseDataItem(),
+		value:        b,
+		locked:       false,
+	}
+}
+
+func (b *Bool) String() string {
+	if b.value {
+		return "true"
+	}
+	return "false"
+}
+
+// SetBool - makes the Bool dataItem implement 2 way binding
+// pass a newvalue to set the dataItem, and set mask to ignore
+// the callback with the masked ID.  Pass a mask of 0 to trigger
+// all callbacks.
+func (b *Bool) SetBool(newvalue bool, mask int) {
+	if b.locked {
```

Comment:
This pattern feels dodgy.
We will be ignoring updates that come in after other updates. Should they not be queued?
Should the lock (b.Lock(), b.Unlock()) not take care of this?

Reply:
Yes, its a hack, and it doesnt feel right to need to do this.

Its a simple solution to a problem that is maybe specific to the implementation ... added to give me a working demo.

I would love to remove it, but need to do this with care, after a better design is drafted.